### PR TITLE
bosun: capturing timings around template rendering.

### PR DESCRIPTION
**Pull request by captncraig**

### CODE REVIEW

File: cmd/bosun/sched/check.go [hunk 1]
```diff
@@ -132,40 +133,8 @@ func (s *Schedule) RunHistory(r *RunHistory) {
...
-					emailbody = body
-					attachments = nil
-				}
-				state.Subject = string(subject)
-				state.Body = string(body)
-				state.EmailBody = emailbody
-				state.EmailSubject = emailsubject
-				state.Attachments = attachments
-			}
+
```
Comment (kylebrandt) [inline cmd/bosun/sched/check.go]: No blank lines within funcs

File: cmd/bosun/sched/check.go [hunk 2]
```diff
@@ -250,6 +219,61 @@ func (s *Schedule) RunHistory(r *RunHistory) {
...
 
+func (s *Schedule) executeTemplates(state *State, event *Event, a *conf.Alert, r *RunHistory) {
+	state.Subject = ""
+	state.Body = ""
+	state.EmailBody = nil
+	state.EmailSubject = nil
+	state.Attachments = nil
+	if event.Status != StUnknown {
+		metric := "template.render"
+
```
Comment (kylebrandt) [inline cmd/bosun/sched/check.go]: Blank lines again... won't comment on the rest of them

File: collect/collect.go [hunk 3]
```diff
@@ -289,6 +289,21 @@ func Sample(metric string, ts opentsdb.TagSet, v float64) error {
...
 }
 
+// StartTimer records the current time, and returns a function you can call to
+// record the end of your action.
+//
+// Typical usage would be:
+//    done := collect.StartTimer("myMetric", opentsdb.TagSet{})
+//    doMyThing()
+//    done()
+func StartTimer(metric string, ts opentsdb.TagSet) func() {
```
Comment (kylebrandt) [inline collect/collect.go]: Consider testing.Benchmark (http://stackoverflow.com/questions/8350609/how-do-you-time-a-function-in-go-and-return-its-runtime-in-milliseconds). Haven't looked into this in depth, maybe using testing here is just wrong. Up to you

**Comment by kylebrandt** [comment]
Also remember commits should be cmd/bosun prefixed

File: collect/collect.go (hunk 3 above)
Comment (captncraig) [inline collect/collect.go]: Benchmark tests are great for build-time benchmarks. It can't be used for run-time instrumentation like this is doing.

**Pull request merged**